src/main/java/uk/gov/digital/ho/hocs/audit/AuditDataService.java

```diff
@@ -2,27 +2,98 @@
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.audit.auditdetails.dto.CreateAuditDto;
 import uk.gov.digital.ho.hocs.audit.auditdetails.model.AuditData;
 import uk.gov.digital.ho.hocs.audit.auditdetails.repository.AuditRepository;
 
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+
+import java.util.List;
+import java.util.UUID;
 
 @Service
 @Slf4j
 public class AuditDataService {
 
     private final AuditRepository auditRepository;
 
+    private Pageable pageRequest;
+
     @Autowired
     public AuditDataService(AuditRepository auditRepository){
         this.auditRepository = auditRepository;
     }
 
-    public AuditData createAudit(String correlationID, String raisingService, String before, String after, String namespace, String type, String userID){
-        log.debug("Creating Audit: {}, ID: {}, Raised by: {}, By user: {}", correlationID, raisingService, userID);
-        AuditData auditData = new AuditData(correlationID, raisingService, before, after, namespace, type, userID);
+    public AuditData createAudit(CreateAuditDto createAuditDto) {
+        AuditData auditData = AuditData.fromDto(createAuditDto);
         auditRepository.save(auditData);
-        log.info("Creating Audit: {}, ID: {}, Raised by: {}, By user: {}", correlationID, raisingService, userID);
+        log.info("Created Audit: UUID: {}, Correlation ID: {}, Raised by: {}, Payload: {}, By user: {}, at timestamp: {}",
+                auditData.getUuid(),
+                auditData.getCorrelationID(),
+                auditData.getRaisingService(),
+                auditData.getAuditPayload(),
+                auditData.getUserID(),
+                auditData.getAuditTimestamp());
         return auditData;
     }
+
+    public AuditData getAuditDataByUUID(UUID auditUUID) {
+        log.info("Requesting Audit for Audit UUID: {} ", auditUUID);
+        return auditRepository.findAuditDataByUuid(auditUUID);
+    }
+
+    public List<AuditData> getAuditDataList(int page, int limit){
+        log.info("Requesting all audits from last seven days");
+        pageRequest = PageRequest.of(page,limit);
+        LocalDateTime lastWeek = LocalDateTime.now().minusWeeks(1);
+        return auditRepository.findAuditData(lastWeek, pageRequest);
+    }
+
+    public List<AuditData> getAuditDataByDateRange(String fromDate, String toDate, int page, int limit){
+        log.info("Requesting all audits for dates: {} to {} ", fromDate, toDate);
+        pageRequest = PageRequest.of(page,limit);
+        return auditRepository.findAuditDataByDateRange(convertLocalDateToStartOfLocalDateTime(fromDate), convertLocalDateToEndOfLocalDateTime(toDate), pageRequest);
+    }
+
+    public List<AuditData> getAuditDataByCorrelationID(String correlationID, int page, int limit){
```

Comment:
for correlation ID you shouldn't need the date range filter. Correlation IDs relate to a single transaction so will only span a few minutes at most.